Adds Create, Delete, Retention (#2020)
temporalio/documentation: docs-src/cloud/tcld/namespace/retention.md

---
id: retention
title: tcld namespace retention
sidebar_label: retention
description: How to manage the length of time a closed Workflow is preserved before deletion.
tags:
  - tcld
---

> The `tcld namespace retention` command manages the length of time (in days) a closed Workflow is preserved before deletion for a given Namespace in Temporal Cloud.

Reviewer: Should "Workflow" be "Workflow Execution" (here and two following instances)?

> The `tcld namespace retention` command manages the length of time (in days) a closed Workflow is preserved before deletion for a given Namespace in Temporal Cloud.

Reviewer: ```suggestion
The `tcld namespace retention` commands manage the length of time (in days) a closed Workflow is preserved before deletion for a given Namespace in Temporal Cloud.
```

Alias: `r`

`tcld namespace retention`

> The following modifier controls the behavior of the command.

Reviewer: `set` and `get` are commands instead of modifiers. Suggest restructuring similar to `tcld namespace search-attributes`, giving `set` and `add` separate Markdown files and specifying the modifiers. (Conveniently, `--namespace` can be a straightforward copy-and-paste.)

### `set`

> Set the length of time (in days) a closed Workflow will be preserved before deletion for a given namespace.

Reviewer: ```suggestion
Set the length of time (in days) a closed Workflow will be preserved before deletion for the specified Namespace.
```

Alias: `s`

**Example**

```bash
tcld namespace retention set --namespace <namespace_id> --retention-days <retention_days>
```

### `get`

> Retrieve the length of time (in days) a closed Workflow will be preserved before deletion for a given namespace.

Reviewer: ```suggestion
Retrieve the length of time (in days) a closed Workflow will be preserved before deletion for the specified Namespace.
```
